controllers/storagecluster/provider_server.go

```diff
@@ -448,12 +448,14 @@ func RandomString(l int) string {
 
 func getOnboardingJobObject(instance *ocsv1.StorageCluster) *batchv1.Job {
 
+	ptrToZero := int32(0)
 	return &batchv1.Job{
 		ObjectMeta: metav1.ObjectMeta{
 			Name:      onboardingJobName,
 			Namespace: instance.Namespace,
 		},
 		Spec: batchv1.JobSpec{
+			TTLSecondsAfterFinished: &ptrToZero, //Eligible to delete automatically when job completes
```

Comment:
No need to have a local variable. 
This is a common case and I think we had a util for it. If not please add the following util (in the proper file in the utils package)

```
func ToPointer[T any](value T) *T {
    return &T
}
``` 

This will allow you to do
```suggestion
			TTLSecondsAfterFinished: utils.ToPointer(0)
```